packages/api/src/cli.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Result } from "neverthrow";
+import { FxError } from "./error";
+
+export type OptionValue = string | boolean | string[] | undefined;
+export type CLIOptionType = "boolean" | "string" | "array";
+
+export interface CLICommand {
+  name: string;
+  fullName?: string;
+  version?: string;
```

Comment:
What does `version` and `fullName` point to?

Reply:
Version is the CLI command version, only root node need this
fullName is like: "teamsfx new template"